benchmark/framework/parser/file_access.go

```diff
@@ -0,0 +1,177 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package bparser
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+var (
+	outputDir           = "./output/"
+	FileAccessDir       = outputDir + "file_access_logs/"
+	sociLogs            = outputDir + "soci-snapshotter-stderr"
+	containerdLogs      = outputDir + "containerd-stderr"
+	sociLogsFuseMessage = "FUSE operation"
+)
+
+type FileAccessPatterns struct {
+	ImageName           string         `json:"ImageName"`
+	ContainerStartTime  time.Time      `json:"containerStartTime"`
+	TotalOperationCount map[string]int `json:"TotalOperationCounts"`
+	Operations          []Operation    `json:"operations"`
+}
+
+type SociLog struct {
+	Msg string `json:"msg"`
+}
+type BaseOperation struct {
+	Level     string    `json:"level"`
+	Msg       string    `json:"msg"`
+	Operation string    `json:"operation"`
+	Path      string    `json:"path"`
+	Time      time.Time `json:"time"`
+}
+
+type Operation struct {
+	Operation                 string `json:"operation"`
+	Path                      string `json:"path"`
+	FirstAccessTimeAfterStart string `json:"firstAccessTimeAfterStart"`
+	Count                     int    `json:"count"`
+}
+
+func ParseFileAccesses(imageName string) error {
+	startTime, err := getTaskStartTime()
+	if err != nil {
+		return err
+	}
+	totalCounts := make(map[string]int)
+	fa := FileAccessPatterns{
+		ImageName:           imageName,
+		ContainerStartTime:  *startTime,
+		TotalOperationCount: totalCounts,
+	}
+	file, err := os.Open(sociLogs)
+	if err != nil {
+		return err
+	}
+
+	defer file.Close()
+
+	b, err := io.ReadAll(file)
```

Comment:
nit: a more idiomatic way to do this is to use a `bufio.Scanner` (https://pkg.go.dev/bufio#Scanner)

```
scanner := bufio.NewScanner(file)
for scanner.Scan() {
    log := scanner.Text()
}
```